### PR TITLE
(SIMP-9720) Fix sssd hooks

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Ensure that the type of LDAP server is specified

SIMP-9720 #close

### CODE REVIEW

**Review by jeannegreulich — COMMENTED** [review]
In the tests that are failing it appears that in fips mode the el8 server cannot connect to the LDAP server to get a list of users and therefor does not create the home directory to mount.  Still trying to figure out what needs to be set to get it working.

**Comment by jeannegreulich** [comment]
Ok, so default openssl policy in el8 for FIPS mode has only TLSv1_3 ciphers.  el7 openssl can't use TLS1_3 because it is not available in the version of openssl.  We dont' want to edit the FIPS policy on el8 and we can't add the ciphers to el7.  

There is something called min_version but it doesn't seem to override the cipher suite, it may just exclude ciphers if you have older ones in the cipher suites.

For some reason if you pass tls_options to net::ldap it is enabling SSL verification in openssl and the ca certs are not loaded into the system certificate store so you have to tell it where the ca cert path is.

conn_opts[:encryption] = { :method =>:start_tls, :tls_options => { :ciphers => ciphers_list, :ca_path => '/etc/pki/simp-testing/pki/cacerts' }}  seems to work.

**Comment by trevor-vaughan** [comment]
@jeannegreulich Thanks for the updates, these look good to me!

File: manifests/create_home_dirs.pp [hunk 1]
```diff
@@ -69,21 +69,25 @@
...
-  Simplib::Syslog::CFacility     $syslog_facility       = 'LOG_LOCAL6',
-  Simplib::Syslog::CSeverity     $syslog_severity       = 'LOG_NOTICE',
-  Boolean                        $strip_128_bit_ciphers = true,
-  Array[String[1]]               $tls_cipher_suite      = simplib::lookup('simp_options::openssl::cipher_suite', { 'default_value' => ['DEFAULT','!MEDIUM'] }),
-  String                         $package_ensure        = simplib::lookup('simp_options::package_ensure', { 'default_value' => 'installed' })
+  Array[Simplib::URI]            $uri                     = simplib::lookup('simp_options::ldap::uri'),
+  String                         $base_dn                 = simplib::lookup('simp_options::ldap::base_dn'),
+  String                         $bind_dn                 = simplib::lookup('simp_options::ldap::bind_dn'),
+  String                         $bind_pw                 = simplib::lookup('simp_options::ldap::bind_pw'),
+  Variant[Enum['simp'],Boolean]  $pki                     = simplib::lookup('simp_options::pki', { 'default_value' => false }),
```
Comment (lnemsick-simp) [inline manifests/create_home_dirs.pp:76]: Please update the CHANGELOG and REFERENCE.md for the new class parameters

**Review by lnemsick-simp — CHANGES_REQUESTED** [review]
docs update

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -357,31 +334,6 @@ pup7.x-unit:
 # Repo-specific content
 # ==============================================================================
```
Comment (lnemsick-simp) [inline .gitlab-ci.yml:335]: Puppet 5 removal should be noted in the changelog

**Review by lnemsick-simp — APPROVED** [review]

**Review by lnemsick-simp — APPROVED** [review]

**Pull request merged**